packages/transactions/src/transaction-size.ts

```diff
@@ -0,0 +1,85 @@
+import { SOLANA_ERROR__TRANSACTION__EXCEEDS_SIZE_LIMIT, SolanaError } from '@solana/errors';
+
+import { getTransactionEncoder } from './codecs';
+import { Transaction } from './transaction';
+
+/**
+ * The maximum size of a transaction packet in bytes.
+ */
+export const TRANSACTION_PACKET_SIZE = 1280;
+
+/**
+ * The size of the transaction packet header in bytes.
+ * This includes the IPv6 header and the fragment header.
+ */
+export const TRANSACTION_PACKET_HEADER =
+    40 /* 40 bytes is the size of the IPv6 header. */ + 8; /* 8 bytes is the size of the fragment header. */
+
+/**
+ * The maximum size of a transaction in bytes.
+ *
+ * Note that this excludes the transaction packet header.
+ * In other words, this is how much content we can fit in a transaction packet.
+ */
+export const TRANSACTION_SIZE_LIMIT = TRANSACTION_PACKET_SIZE - TRANSACTION_PACKET_HEADER;
+
+/**
+ * Gets the size of a given transaction in bytes.
+ *
+ * @example
+ * ```ts
+ * const transactionSize = getTransactionSize(transaction);
+ * ```
+ */
+export function getTransactionSize(transaction: Transaction): number {
+    return getTransactionEncoder().getSizeFromValue(transaction);
+}
+
+/**
+ * A type guard that checks if a transaction is under the size limit.
+ */
+export type TransactionUnderSizeLimit = { readonly __under_size_limit: unique symbol };
+
+/**
+ * Checks if a transaction is under the size limit.
+ *
+ * @typeParam TTransaction - The type of the given transaction.
+ *
+ * @example
+ * ```ts
+ * if (isValidTransactionSize(transaction)) {
+ *    transaction satisfies TransactionUnderSizeLimit;
+ * }
+ * ```
+ */
+export function isValidTransactionSize<TTransaction extends Transaction>(
+    transaction: TTransaction,
+): transaction is TransactionUnderSizeLimit & TTransaction {
```

Comment:
Technically not ‘under’ because of that `<=`

Reply:
`TransactionUnderSizeLimitPlusOne` haha. Any suggestions for something more accurate? Maybe `TransactionWithinSizeLimit`?

Reply:
Within would work. Or `TransactionOfAcceptableSize` or something like that with ‘of.’

Reply:
I have a slight preference for within since `isTransactionWithinSizeLimit` reads quite well IMO but happy to go with something else. ☺️